diff --git a/c-embed.c b/c-embed.c
index e233d4a..2b34a7e 100644
--- a/c-embed.c
+++ b/c-embed.c
@@ -23,16 +23,36 @@ FILE* fs = NULL;    // Virtual Filesystem
 FILE* file = NULL;  // Embed Target File Pointer
 u_int32_t pos = 0;  // Current Position
 
-void cembed(char* filename){
+void cembed(const char* const filename, const char* root_dir){
 
   file = fopen(filename, "rb");  // Open the Embed Target File
   if(file == NULL){
     printf("Failed to open file %s.", filename);
     return;
   }
+  u_int32_t filename_hash = hash(filename);
+
+  if(root_dir != NULL){
+    
+    const char* filename_relative = filename;
+    const size_t filename_len = strlen(filename);
+    const size_t root_len = strlen(root_dir);
+    if(filename_len < root_len){
+      fprintf(stderr, "Invalid file root: %s\n",root_dir);
+      exit(2);
+    }
+    for(size_t i = 0; i < root_len; ++i){
+      if(filename[i] == root_dir[i]){
+        filename_relative++;
+      }
+    }
+
+    filename_hash = hash(filename_relative);
+  }
+
 
   fseek(file, 0, SEEK_END);     // Define Map
-  EMAP map = {hash(filename), pos, (u_int32_t)ftell(file)};
+  EMAP map = {filename_hash, pos, (u_int32_t)ftell(file)};
   rewind (file);
 
   char* buf = malloc(sizeof(char)*(map.size));
@@ -61,14 +81,15 @@ void cembed(char* filename){
 #define CEMBED_DIRENT_DIR 4
 #define CEMBED_MAXPATH 512
 
-void iterdir(char* d){
+void iterdir(const char* const d, const char* root_dir){
 
   char* fullpath = (char*)malloc(CEMBED_MAXPATH*sizeof(char));
 
   DIR *dir;
   struct dirent *ent;
-
-  if ((dir = opendir(d)) != NULL) {
+  dir = opendir(d);
+  
+  if (dir != NULL) {
 
     while ((ent = readdir(dir)) != NULL) {
 
@@ -79,14 +100,14 @@ void iterdir(char* d){
         strcpy(fullpath, d);
         strcat(fullpath, "/");
         strcat(fullpath, ent->d_name);
-        cembed(fullpath);
+        cembed(fullpath, root_dir);
       }
 
       else if(ent->d_type == CEMBED_DIRENT_DIR){
         strcpy(fullpath, d);
         strcat(fullpath, "/");
         strcat(fullpath, ent->d_name);
-        iterdir(fullpath);
+        iterdir(fullpath, root_dir);
       }
 
     }
@@ -98,7 +119,7 @@ void iterdir(char* d){
   else {
 
     strcpy(fullpath, d);
-    cembed(fullpath);
+    cembed(fullpath, root_dir);
 
   }
 
@@ -106,6 +127,28 @@ void iterdir(char* d){
 
 }
 
+static bool is_directory(const char* file){
+  DIR *dir = opendir(file);
+  if (dir != NULL) {
+    closedir(dir);
+    return true;
+  }
+
+  return false;
+}
+
+static void iterdir_start(const char* const d){
+
+  if(is_directory(d)){
+    iterdir(d, d);
+    return;
+  }
+
+  iterdir(d, NULL);
+
+
+}
+
 void system_checked(const char *command){
 
   int result = system(command);
@@ -123,8 +166,10 @@ int main(int argc, char* argv[]){
 
   char fmt[CEMBED_MAXPATH];
 
-  if(argc <= 1)
-    return 0;
+  if(argc <= 1){
+    fprintf(stderr, "Invalid amount of arguments: %d\n", argc);
+    return 1;
+  }
 
   sprintf(fmt, "if [ ! -d %s ]; then mkdir %s; fi;", CEMBED_TMPDIR, CEMBED_TMPDIR);
   system(fmt);
@@ -140,7 +185,7 @@ int main(int argc, char* argv[]){
   }
 
   for(int i = 1; i < argc; i++)
-    iterdir(argv[i]);
+    iterdir_start(argv[i]);
 
   fclose(ms);
   fclose(fs);
diff --git a/c-embed.h b/c-embed.h
index 767d5ca..b6a6ba8 100644
--- a/c-embed.h
+++ b/c-embed.h
@@ -17,7 +17,7 @@
 #include <stdbool.h>
 #include <string.h>
 
-u_int32_t hash(char * key){   // Hash Function: MurmurOAAT64
+u_int32_t hash(const char * key){   // Hash Function: MurmurOAAT64
   u_int32_t h = 3323198485ul;
   for (;*key;++key) {
     h ^= *key;
