diff --git a/meson.build b/meson.build
index e82ae84..a42ad48 100644
--- a/meson.build
+++ b/meson.build
@@ -65,6 +65,13 @@ if brotli_found_all
   args += '-DCPPHTTPLIB_BROTLI_SUPPORT'
 endif
 
+cpp = meson.get_compiler('cpp')
+if cpp.get_id() == 'gcc' and host_machine.system() == 'windows'
+  deps += cpp.find_library('wsock32')
+  deps += cpp.find_library('ws2_32')
+  deps += cpp.find_library('crypt32')
+endif
+
 cpp_httplib_dep = dependency('', required: false)
 
 if get_option('cpp-httplib_compile')
