diff --git a/meson.build b/meson.build
index 45cd1d1..13b276d 100644
--- a/meson.build
+++ b/meson.build
@@ -1,10 +1,13 @@
 project('fmt', 'cpp', version: '10.2.0', license: 'MIT WITH fmt-exception', default_options: ['cpp_std=c++14'])
 
+#TODO: remove with the next release
+cpp = meson.get_compiler('cpp')
+
 fmt_private_cpp_args = []
 fmt_interface_cpp_args = []
 if get_option('default_library') == 'shared'
   fmt_private_cpp_args += ['-DFMT_LIB_EXPORT']
-  fmt_interface_cpp_args += ['-DFMT_SHARED']
+  fmt_interface_cpp_args += ['-DFMT_SHARED', cpp.get_supported_arguments('-Wno-tautological-compare'),]
 endif
 
 fmt_lib = library(
