diff --git a/meson.build b/meson.build
index 0c841f7..0c698c6 100644
--- a/meson.build
+++ b/meson.build
@@ -27,10 +27,6 @@ fmt_dep = declare_dependency(
   link_with: fmt_lib,
 )
 
-if meson.version().version_compare('>=0.54.0')
-  meson.override_dependency('fmt', fmt_dep)
-endif
-
 fmt_header_only_dep = declare_dependency(
   include_directories: 'include',
   compile_args: '-DFMT_HEADER_ONLY',
