diff --git a/meson.build b/meson.build
index c231c7f..66f96ac 100644
--- a/meson.build
+++ b/meson.build
@@ -4,7 +4,7 @@ project(
     version: '0.9.5',
 )
 
-magic_enum_include = include_directories('include/magic_enum')
+magic_enum_include = include_directories('include')
 
 magic_enum_args = []
 
@@ -20,3 +20,18 @@ magic_enum_dep = declare_dependency(
 if get_option('test')
     subdir('test')
 endif
+
+install_headers(
+    files(
+        'include/magic_enum/magic_enum.hpp',
+        'include/magic_enum/magic_enum_all.hpp',
+        'include/magic_enum/magic_enum_containers.hpp',
+        'include/magic_enum/magic_enum_flags.hpp',
+        'include/magic_enum/magic_enum_format.hpp',
+        'include/magic_enum/magic_enum_fuse.hpp',
+        'include/magic_enum/magic_enum_iostream.hpp',
+        'include/magic_enum/magic_enum_switch.hpp',
+        'include/magic_enum/magic_enum_utility.hpp',
+    ),
+    subdir: 'magic_enum',
+)
