diff --git a/meson.build b/meson.build
index 68eb17b..f147500 100644
--- a/meson.build
+++ b/meson.build
@@ -855,6 +855,7 @@ if platform_is_android
     extra_deps += [
         cc.find_library('android'),
         cc.find_library('log'),
+        cc.find_library('OpenSLES'),
     ]
     if not opt_video_openglesv2.disabled()
         extra_deps += cc.find_library('GLESv2', required : opt_video_openglesv2.enabled())
diff --git a/src/core/android/meson.build b/src/core/android/meson.build
index 8c4338b..3bd041f 100644
--- a/src/core/android/meson.build
+++ b/src/core/android/meson.build
@@ -1,3 +1,4 @@
 c_sources += files(
     'SDL_android.c',
+    'cpu-features.c'
 )
