diff --git a/meson.build b/meson.build
index 6d49243..e095a61 100644
--- a/meson.build
+++ b/meson.build
@@ -1190,8 +1190,7 @@ endif
 sdl2_dep = declare_dependency(
     link_with : sdl2,
     dependencies : deps,
-    include_directories : core_inc,
-    compile_args : c_args,
+    include_directories : core_inc
 )
 
 test_deps = []
@@ -1215,8 +1214,7 @@ sdl2_test = static_library('sdl2_test',
 sdl2_test_dep = declare_dependency(
     link_with : sdl2_test,
     dependencies : [sdl2_dep],
-    include_directories : core_inc,
-    compile_args : c_args,
+    include_directories : core_inc
 )
 
 meson.override_dependency('sdl2', sdl2_dep)
