diff --git a/meson.build b/meson.build
index d8f0318..5bd50e5 100644
--- a/meson.build
+++ b/meson.build
@@ -20,6 +20,13 @@ if fmt_dep.found()
   spdlog_compile_args += '-DSPDLOG_FMT_EXTERNAL'
 endif
 
+if host_machine.system() == 'android'
+  cpp = meson.get_compiler('cpp')
+  spdlog_dependencies += [
+        cpp.find_library('log')
+    ]
+endif
+
 if get_option('compile_library')
   spdlog_compile_args += '-DSPDLOG_COMPILED_LIB'
   if get_option('default_library') != 'static'
