diff --git a/src/meson.build b/src/meson.build
index 969963c..dc11467 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -13,12 +13,16 @@ endif
 
 version_parts = meson.project_version().split('.')
 
+cpp = meson.get_compiler('cpp')
+private_arguments = cpp.get_supported_arguments('/IGNORE:4275')
+
+
 spdlog_lib = library(
   'spdlog',
   src,
   include_directories: inc,
   dependencies: spdlog_dependencies,
-  cpp_args: spdlog_compile_args,
+  cpp_args: spdlog_compile_args + private_arguments,
   soversion: version_parts[0] + '.' + version_parts[1],
   version: meson.project_version(),
   install: true,
