diff --git a/lib/curl_setup.h b/lib/curl_setup.h
index 7a6ef10..bbcc071 100644
--- a/lib/curl_setup.h
+++ b/lib/curl_setup.h
@@ -1019,4 +1019,12 @@ int getpwuid_r(uid_t uid, struct passwd *pwd, char *buf,
 #  define CURL_INLINE /* empty */
 #endif
 
+#if defined(__SWITCH__)
+
+# undef HAVE_TERMIOS_H
+# undef HAVE_PIPE
+
+#endif
+
+
 #endif /* HEADER_CURL_SETUP_H */
diff --git a/meson.build b/meson.build
index 65c9374..8bd044e 100644
--- a/meson.build
+++ b/meson.build
@@ -140,6 +140,18 @@ if host_machine.system() == 'darwin'
   )
 endif
 
+if meson.is_cross_build()
+  if host_machine.system() == 'switch'
+    sys_deps += cc.find_library(
+      'nx',
+      required: false,
+    )
+  else
+    error('Unhandled cross built system: ' + host_machine.system())
+  endif
+endif
+
+
 threads_dep = dependency(
   'threads',
   required: false,
