diff --git a/MdePkg/Include/Library/DebugLib.h b/MdePkg/Include/Library/DebugLib.h
index 30cfcc77f9..449b11d9f1 100644
--- a/MdePkg/Include/Library/DebugLib.h
+++ b/MdePkg/Include/Library/DebugLib.h
@@ -131,6 +131,10 @@ SPDX-License-Identifier: BSD-2-Clause-Patent
                       based on the format string specified by Format.
 
 **/
+#if defined(__cplusplus)
+extern "C" {
+#endif
+
 VOID
 EFIAPI
 DebugPrint (
@@ -669,3 +673,7 @@ UnitTestDebugAssert (
     NULL :                                                                                  \
     BASE_CR (Record, TYPE, Field)
 #endif
+
+#if defined(__cplusplus)
+  }
+#endif
