diff --git a/src/settings.js b/src/settings.js
index 981c44fa..17ca0078 100644
--- a/src/settings.js
+++ b/src/settings.js
@@ -1624,7 +1624,7 @@ var SDL2_IMAGE_FORMATS = [];
 
 // Formats to support in SDL2_mixer. Valid values: ogg, mp3, mod, mid
 // [link]
-var SDL2_MIXER_FORMATS = ["ogg"];
+var SDL2_MIXER_FORMATS = [];
 
 // 1 = use sqlite3 from emscripten-ports
 // Alternate syntax: --use-port=sqlite3
