diff --git a/tools/ports/sdl2_image.py b/tools/ports/sdl2_image.py
index 138fb3f..9a3d2e6 100644
--- a/tools/ports/sdl2_image.py
+++ b/tools/ports/sdl2_image.py
@@ -16,14 +16,16 @@ variants = {
 
 OPTIONS = {
   'formats': 'A comma separated list of formats (ex: --use-port=sdl2_image:formats=png,jpg)',
+  'mt': 'use pthread'
 }
 
 SUPPORTED_FORMATS = {'bmp', 'gif', 'jpg', 'lbm', 'pcx', 'png',
                      'pnm', 'qoi', 'svg', 'tga', 'xcf', 'xpm', 'xv'}
 
 # user options (from --use-port)
-opts: dict[str, set] = {
+opts: dict[str, set | int] = {
   'formats': set(),
+  'mt': 0
 }
 
 
@@ -41,7 +43,7 @@ def get_lib_name(settings):
   libname = 'libSDL2_image'
   if formats:
     libname += '-' + formats
-  if settings.PTHREADS:
+  if settings.PTHREADS or opts['mt']:
     libname += '-mt'
   if settings.SUPPORT_LONGJMP == 'wasm':
     libname += '-wasm-sjlj'
@@ -71,7 +73,7 @@ def get(ports, settings, shared):
     if 'jpg' in formats:
       flags += ['-sUSE_LIBJPEG']
 
-    if settings.PTHREADS:
+    if settings.PTHREADS or opts['mt']:
       flags += ['-pthread']
 
     if settings.SUPPORT_LONGJMP == 'wasm':
@@ -106,6 +108,12 @@ def handle_options(options, error_handler):
     else:
       opts['formats'].add(format)
 
+  mt = options['mt']
+  if mt not in ["1", "0"]:
+    error_handler(f'{mt} has to be either 0 or 1')
+  else:
+    opts['mt'] = int(mt)
+
 
 def show():
   return 'sdl2_image (-sUSE_SDL_IMAGE=2 or --use-port=sdl2_image; zlib license)'
