diff --git a/tools/ports/sdl2_mixer.py b/tools/ports/sdl2_mixer.py
index f77f906d..417b2a79 100644
--- a/tools/ports/sdl2_mixer.py
+++ b/tools/ports/sdl2_mixer.py
@@ -14,14 +14,27 @@ variants = {
   'sdl2_mixer_none': {'SDL2_MIXER_FORMATS': []},
 }
 
+OPTIONS = {
+  'formats': 'A comma separated list of formats (ex: --use-port=sdl2_mixer:formats=mp3)'
+}
+
+SUPPORTED_FORMATS = {'mp3', 'ogg', 'mod', 'mid'}
+
+# user options (from --use-port)
+opts = {
+  'formats': set(),
+}
 
 def needed(settings):
   return settings.USE_SDL_MIXER == 2
 
+def get_formats(settings):
+  return set(settings.SDL2_MIXER_FORMATS).union(opts['formats'])
+
+
 
 def get_lib_name(settings):
-  settings.SDL2_MIXER_FORMATS.sort()
-  formats = '-'.join(settings.SDL2_MIXER_FORMATS)
+  formats = '-'.join(sorted(get_formats(settings)))
 
   libname = 'libSDL2_mixer'
   if formats != '':
@@ -44,26 +57,28 @@ def get(ports, settings, shared):
       '-O2',
       '-DMUSIC_WAV',
     ]
+    
+    formats = get_formats(settings)
 
-    if "ogg" in settings.SDL2_MIXER_FORMATS:
+    if "ogg" in formats:
       flags += [
         '-sUSE_VORBIS',
         '-DMUSIC_OGG',
       ]
 
-    if "mp3" in settings.SDL2_MIXER_FORMATS:
+    if "mp3" in formats:
       flags += [
         '-sUSE_MPG123',
         '-DMUSIC_MP3_MPG123',
       ]
 
-    if "mod" in settings.SDL2_MIXER_FORMATS:
+    if "mod" in formats:
       flags += [
         '-sUSE_MODPLUG',
         '-DMUSIC_MOD_MODPLUG',
       ]
 
-    if "mid" in settings.SDL2_MIXER_FORMATS:
+    if "mid" in formats:
       flags += [
         '-DMUSIC_MID_TIMIDITY',
       ]
@@ -104,16 +119,29 @@ def clear(ports, settings, shared):
 
 def process_dependencies(settings):
   settings.USE_SDL = 2
-  if "ogg" in settings.SDL2_MIXER_FORMATS:
+  
+  formats = get_formats(settings)
+  
+  if "ogg" in formats:
     deps.append('vorbis')
     settings.USE_VORBIS = 1
-  if "mp3" in settings.SDL2_MIXER_FORMATS:
+  if "mp3" in formats:
     deps.append('mpg123')
     settings.USE_MPG123 = 1
-  if "mod" in settings.SDL2_MIXER_FORMATS:
+  if "mod" in formats:
     deps.append('libmodplug')
     settings.USE_MODPLUG = 1
 
+def handle_options(options, error_handler):
+  formats = options['formats'].split(',')
+  for format in formats:
+    format = format.lower().strip()
+    if format not in SUPPORTED_FORMATS:
+      error_handler(f'{format} is not a supported format')
+    else:
+      opts['formats'].add(format)
+      
+
 
 def show():
   return 'sdl2_mixer (-sUSE_SDL_MIXER=2 or --use-port=sdl2_mixer; zlib license)'
