diff --git a/tools/ports/sdl2_image.py b/tools/ports/sdl2_image.py
index 70fa1499..36be807b 100644
--- a/tools/ports/sdl2_image.py
+++ b/tools/ports/sdl2_image.py
@@ -18,7 +18,8 @@ variants = {
 }
 
 OPTIONS = {
-  'formats': 'A comma separated list of formats (ex: --use-port=sdl2_image:formats=png,jpg)'
+  'formats': 'A comma separated list of formats (ex: --use-port=sdl2_image:formats=png,jpg)',
+  'mt': 'use pthread'
 }
 
 SUPPORTED_FORMATS = {'avif', 'bmp', 'gif', 'jpg', 'jxl', 'lbm', 'pcx', 'png',
@@ -26,7 +27,8 @@ SUPPORTED_FORMATS = {'avif', 'bmp', 'gif', 'jpg', 'jxl', 'lbm', 'pcx', 'png',
 
 # user options (from --use-port)
 opts: Dict[str, Set] = {
-  'formats': set()
+  'formats': set(),
+  'mt': 0
 }
 
 
@@ -44,7 +46,7 @@ def get_lib_name(settings):
   libname = 'libSDL2_image'
   if formats != '':
     libname += '-' + formats
-  if settings.PTHREADS:
+  if settings.PTHREADS or opts['mt']:
     libname += '-mt'
   return libname + '.a'
 
@@ -75,7 +77,7 @@ def get(ports, settings, shared):
     if 'jpg' in formats:
       flags += ['-sUSE_LIBJPEG']
 
-    if settings.PTHREADS:
+    if settings.PTHREADS or opts['mt']:
       flags += ['-pthread']
 
     ports.build_port(src_dir, final, 'sdl2_image', flags=flags, srcs=srcs)
@@ -106,6 +108,12 @@ def handle_options(options, error_handler):
       error_handler(f'{format} is not a supported format')
     else:
       opts['formats'].add(format)
+      
+  mt = options['mt']
+  if mt not in ["1","0"]:
+    error_handler(f'{mt} has to be either 0 or 1')
+  else:
+    opts['mt'] = int(mt)
 
 
 def show():
