diff --git a/tools/ports/sdl2_image.py b/tools/ports/sdl2_image.py
index c72ef576..0c12feba 100644
--- a/tools/ports/sdl2_image.py
+++ b/tools/ports/sdl2_image.py
@@ -16,15 +16,17 @@ variants = {
 }
 
 OPTIONS = {
-  'formats': 'A comma separated list of formats (ex: --use-port=sdl2_image:formats=png,jpg)'
+  'formats': 'A comma separated list of formats (ex: --use-port=sdl2_image:formats=png,jpg)',
+  'mt': 'use pthread'
 }
 
 SUPPORTED_FORMATS = {'avif', 'bmp', 'gif', 'jpg', 'jxl', 'lbm', 'pcx', 'png',
                      'pnm', 'qoi', 'svg', 'tga', 'tif', 'webp', 'xcf', 'xpm', 'xv'}
 
 # user options (from --use-port)
-opts: Dict[str, Set] = {
-  'formats': set()
+opts = {
+  'formats': set(),
+  'mt': 0
 }
 
 
@@ -42,7 +44,7 @@ def get_lib_name(settings):
   libname = 'libSDL2_image'
   if formats != '':
     libname += '_' + formats
-  return libname + '.a'
+  return libname + ('-mt' if opts['mt'] else '') + '.a'
 
 
 def get(ports, settings, shared):
@@ -70,6 +72,8 @@ def get(ports, settings, shared):
 
     if 'jpg' in formats:
       defs += ['-sUSE_LIBJPEG']
+    if opts['mt']:
+      defs += ['-pthread']
 
     ports.build_port(src_dir, final, 'sdl2_image', flags=defs, srcs=srcs)
 
@@ -99,7 +103,12 @@ def handle_options(options, error_handler):
       error_handler(f'{format} is not a supported format')
     else:
       opts['formats'].add(format)
-
+      
+  mt = options['mt']
+  if mt not in ["1","0"]:
+    error_handler(f'{mt} has to be either 0 or 1')
+  else:
+    opts['mt'] = int(mt)
 
 def show():
   return 'sdl2_image (-sUSE_SDL_IMAGE=2 or --use-port=sdl2_image; zlib license)'
